proxies.py

```diff
@@ -0,0 +1,183 @@
+"""
+Websocket proxy server
+
+Copyright 2016 NaviSite Inc - A Time Warner Cable Company
+"""
+
+import websockify
+import urlparse
+import time
+import select
+import sys
+import errno
+import six
+import socket
+
+class SimpleProxyHandler(websockify.ProxyRequestHandler):
+    """
+    Simple proxy that expects both sides to support websockets
+    """
+
+    def _recv_line(self, sock):
+        """
+        Receive from the socket on char at a time
+        until we get to a new line
+
+        Args:
+          sock: The socket to receive from
+        """
+        line = []
+        while True:
+            character = sock.recv(1)
+            line.append(character)
+            if character == six.b("\n"):
+                break
+        return six.b("").join(line)
+
+    def _read_headers(self, sock):
+        """
+        Read the headers line by line, we previously send to the target sock
+        
+        Args:
+          sock: The socket we have send headers to
+        """
+        status = None
+        headers = {}
+
+        while True:
+            line = self._recv_line(sock)
+            line = line.decode('utf-8').strip()
+            if not line:
+                break
+
+            if not status:
```

Comment:
I may be missing something, but I do not see how status will ever be anything but None here. If is initialized to None before the loop, and the only way that it gets set to something different is if it is not None at this point.